src/lib/LibArithmetic.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import { FixedPointMathLib } from "solady/utils/FixedPointMathLib.sol";
+
+/// @title LibArithmetic
+/// @notice General Math
+library LibArithmetic {
+    function clamp(int256 _value, int256 _min, int256 _max) internal pure returns (int256 clampedValue) {
+        return min(max(_value, _min), _max);
+    }
+
+    /// @notice Returns the smallest of two signed numbers.
+    function max(int256 a, int256 b) internal pure returns (int256 result) {
+        assembly ("memory-safe") {
+            let a_is_positive := eq(shr(255, a), 0)
+            let b_is_positive := eq(shr(255, b), 0)
+            switch and(a_is_positive, b_is_positive)
+            case 1 {
+                result := b
+                if gt(a, b) { result := a }
+            }
+            default {
+                result := b
+                if eq(shr(255, a), 0) { result := a }
+            }
+        }
+    }
+
+    /// @notice Returns the smallest of two signed numbers.
+    function min(int256 a, int256 b) internal pure returns (int256 result) {
+        assembly ("memory-safe") {
+            let a_is_positive := eq(shr(255, a), 0)
+            let b_is_positive := eq(shr(255, b), 0)
+            switch and(a_is_positive, b_is_positive)
+            case 1 {
+                result := b
+                if lt(a, b) { result := a }
+            }
+            default {
+                result := b
+                if eq(shr(255, b), 0) { result := a }
+            }
+        }
+    }
```

Comment:
The logic in these functions seems to be a bit flawed, the following tests both fail:
```solidity
    function testDiff_min(int256 a, int256 b) public {
        assertEq(a < b ? a : b, LibArithmetic.min(a, b));
    }

    function testDiff_max(int256 a, int256 b) public {
        assertEq(a > b ? a : b, LibArithmetic.max(a, b));
    }
```

Solady's `FixedPointMathLib` already has a `min` and `max` function for signed integers that's branchless + a `clamp` function, we should only need `cdexp` 😄